File: android/app/src/main/java/com/openclaw/console/data/model/RecurringTask.kt
package com.openclaw.console.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.json.JsonElement

@Serializable
data class RecurringTaskSchedule(
    val type: String, // "cron" or "interval"
    val value: JsonElement
)

@Serializable
data class RecurringTask(
    val id: String,
    @SerialName("agent_id") val agentId: String,
    val name: String,
    val description: String,
    val schedule: RecurringTaskSchedule,
    @SerialName("last_run") val lastRun: String?,
    @SerialName("next_run") val nextRun: String?,
    val status: String,
    @SerialName("error_count") val errorCount: Int
)

File: ios/OpenClawConsole/OpenClawConsole/Models/RecurringTask.swift
// Models/RecurringTask.swift
// OpenClaw Work Console
// Model for background loops/recurring tasks

import Foundation

struct RecurringTask: Codable, Identifiable, Hashable {
    let id: String
    let name: String
    let description: String?
    let schedule: String
    let enabled: Bool
    let lastRun: Date?
    let nextRun: Date?
    let status: String

    enum CodingKeys: String, CodingKey {
        case id
        case name
        case description
        case schedule
        case enabled
        case lastRun = "last_run"
        case nextRun = "next_run"
        case status
    }
}